simulator/tests/randpa_finality_tests.cpp

```diff
@@ -519,3 +542,80 @@ TEST(randpa_finality, randpa_disabled_nodes) {
         EXPECT_EQ(3994, get_block_height(runner.get_db(i).last_irreversible_block_id()));
     }
 }
+
+TEST(randpa_finality, large_multisig_BP_only) {
+    constexpr size_t n_nodes = 6;
+    constexpr size_t sig_provs_per_node = 15;
+
+    // TODO: use this for testing set_signature_providers()
+    //~// generate arrays of signature providers & public keys (one array for each physical node)
```

Comment:
I figured out some problems with `set_signature_providers()` function, it should be also tested. (Also it is currently not used anywhere, except simulator, so may be done later.)